src/btree/bt_read.c

```diff
@@ -160,26 +200,22 @@ __page_read(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags)
      * In the case of a fast delete, move all of the page's records to a deleted state based on the
      * fast-delete information. Skip for special commands that don't care about an in-memory state.
      *
-     * Note: there are three possible cases - the state was WT_REF_DELETED and ft_info.del was NULL;
-     * the state was WT_REF_DELETED and ft_info.del was non-NULL; and the state was WT_REF_DISK and
-     * the parent page cell was a WT_CELL_ADDR_DEL cell. The last is only valid in a readonly tree.
+     * There are two possible cases: the state was WT_REF_DELETED and page_del was or wasn't NULL.
+     * It used to also be possible for eviction to set the state to WT_REF_DISK while the parent
+     * page nonetheless had a WT_CELL_ADDR_DEL cell. This is not supposed to happen any more, so for
+     * now at least assert it doesn't.
      *
-     * ft_info.del gets cleared and set to NULL if the deletion is found to be globally visible;
-     * this can happen in any of several places.
+     * page_del gets cleared and set to NULL if the deletion is found to be globally visible; this
+     * can happen in any of several places.
      */
-    del = NULL;
     if (previous_state == WT_REF_DISK) {
```

Comment:
I would change this to:

WT_ASSERT(previous_state != WT_REF_DISK || session, ref->page_del == NULL);
WT_ASSERT(previous_state != WT_REF_DISK || addr.del_set == false);

Reply:
To be consistent within the code base how we write asserts

Reply:
Fixed in [50ff2aa](https://github.com/wiredtiger/wiredtiger/pull/8188/commits/50ff2aa9435ce60b6fddf0cad8611f9602a83e86)